gammapy/spectrum/make.py

```diff
@@ -0,0 +1,220 @@
+        events_region = observation.events.select_region(
+            self.region, wcs=self.geom_ref.wcs
+        )
+        counts.fill(events_region)
+        return counts
+
+    def make_background(self, observation):
+        """Make background
+
+        Parameters
+        ----------
+        observation: `DataStoreObservation`
+            Observation to compute effective area for.
+
+        Returns
+        -------
+        background : `CountsSpectrum`
+            Background spectrum
+        """
+        offset = observation.pointing_radec.separation(self.region.center)
+        energy_hi = self.e_reco[1:]
+        energy_lo = self.e_reco[:-1]
+
+        bkg = observation.bkg
+
+        data = bkg.evaluate_integrate(
+            fov_lon=0 * u.deg, fov_lat=offset, energy_reco=self.e_reco
+        )
+
+        data *= self.region_solid_angle
+        data *= observation.observation_time_duration
+
+        counts = CountsSpectrum(
+            energy_hi=energy_hi, energy_lo=energy_lo, data=data.to_value(""), unit=""
+        )
+        return counts
+
+    def make_aeff(self, observation):
+        """Make effective area
+
+        Parameters
+        ----------
+        observation: `DataStoreObservation`
+            Observation to compute effective area for.
+
+        Returns
+        -------
+        aeff : `EffectiveAreaTable`
+            Effective area table.
+        """
+        offset = observation.pointing_radec.separation(self.region.center)
+        aeff = observation.aeff.to_effective_area_table(offset, energy=self.e_true)
+
+        if self.containment_correction:
+            if not isinstance(self.region, CircleSkyRegion):
+                raise TypeError(
+                    "Containment correction only support for circular regions."
+                )
+            psf = observation.psf
+
+            if isinstance(psf, EnergyDependentMultiGaussPSF):
+                psf = psf.to_psf3d()
+
+            table_psf = psf.to_energy_dependent_table_psf(theta=offset)
+            aeff = apply_containment_fraction(aeff, table_psf, self.region.radius)
+
+        return aeff
+
+    def make_edisp(self, observation):
+        """Make energy dispersion
+
+        Parameters
+        ----------
+        observation: `DataStoreObservation`
+            Observation to compute edisp for.
+
+        Returns
+        -------
+        edisp : `EnergyDispersion`
+            Energy dispersion
+
+        """
+        offset = observation.pointing_radec.separation(self.region.center)
+        edisp = observation.edisp.to_energy_dispersion(
+            offset, e_reco=self.e_reco, e_true=self.e_true
+        )
+        return edisp
+
+    def run(self, observation, selection=None):
+        """Make spectrum dataset.
+
+        Parameters
+        ----------
+        observation: `DataStoreObservation`
+            Observation to reduce.
+        selection : list
+            List of str, selecting which maps to make.
```

Comment:
selecting which `spectral products` or `spectral quantities` to make

Reply:
Fixed